libr/anal/fcn.c

```diff
@@ -2049,3 +2039,13 @@ R_API int r_anal_fcn_count_edges(RAnalFunction *fcn, int *ebbs) {
 	}
 	return edges;
 }
+
+R_API RList *r_anal_fcn_get_refs(RAnal *anal, RAnalFunction *fcn) {
```

Comment:
what about returning const RList and cache and add RAnal.xrefsCacheValid, and do memoization to hold the ownership of that cloned RList, this way the ownership of the list of xrefs is by RAnal, instead of the caller, this way we can speedup some situations, avoid unnecessary alloc/free of refs, etc..

So.. as long as you did that change already, we can also opt for having another getter to not have to manage the xref rlist by hand.

Reply:
but we can do this change later

Reply:
you mean every function caches its refs/xrefs?

Reply:
no, just a single global cache, but nvm, lets go forward with this approach first, and we can evaluate the benefits of caching later. just fix the memleak i pointed out and ill merge